orc/src/main/java/org/apache/iceberg/orc/OrcFileAppender.java

```diff
@@ -108,12 +111,44 @@ public long length() {
   @Override
   public void close() throws IOException {
     if (!isClosed) {
-      this.isClosed = true;
-      writer.close();
+      try {
+        if (batch.size > 0) {
+          writer.addRowBatch(batch);
+          batch.reset();
+        }
+      }
+      finally {
+        writer.close();
+        this.isClosed = true;
+      }
     }
   }
 
   public TypeDescription getSchema() {
     return orcSchema;
   }
+
+  private static Writer newOrcWriter(OutputFile file,
+                                     ColumnIdMap columnIds,
+                                     OrcFile.WriterOptions options, Map<String, byte[]> metadata) {
+    final Path locPath = new Path(file.location());
+    final Writer writer;
+
+    try {
+      writer = OrcFile.createWriter(locPath, options);
```

Comment:
Is it possible to pass an output stream?

Reply:
Unfortunately no, I checked the `OrcFile` signatures and all (`createReader` and `createWriter`) use `Path` only. Having a similar method to this receiving an output stream would be ideal.